### ISSUE TITLE
"check-separator" Stop checking whitespace after last separated item's semicolon

### ISSUE THREAD

**Issue by hapet-js** [palantir/tslint#2147]
![image](https://cloud.githubusercontent.com/assets/20340782/22367991/cc211fe2-e49d-11e6-85a1-efcdfc76d90b.png)

### Bug Report

- __TSLint version__: `4.4.2`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: `Webpack tslint-loader`

#### TypeScript code being linted

```ts
<div onClick={() => {this.setState({showModal: true});}}/>
```

with `tslint.json` configuration:

```json
"whitespace": [true,
        "check-branch",
        "check-decl",
        "check-module",
        "check-operator",
        "check-separator",
        "check-type",
        "check-typecast",
        "check-preblock"
    ],
```

#### Actual behavior

missing whitespace

#### Expected behavior

clean :)

**Comment by JoshuaKGoldberg** [comment]
I guess this is a bug that can be fixed in the rule... though it feels odd having a semicolon at that point. Why bother including it at all?

**Comment by JoshuaKGoldberg** [comment]
Per #4534, closing this as a whitespace/formatting rule. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
